Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -998,16 +997,15 @@ class BrowserViewController: UIViewController {
         urlBar.tabLocationViewDidTapLocation(urlBar.locationView)
     }
 
-    private func popToTab(forTab: Tab? = nil) {
+    private func popToTab() {
         guard let currentViewController = navigationController?.topViewController else {
                 return
         }
         if let presentedViewController = currentViewController.presentedViewController {
             presentedViewController.dismissViewControllerAnimated(false, completion: nil)
         }
         // if a tab already exists and the top VC is not the BVC then pop the top VC, otherwise don't.
-        if currentViewController != self,
-            let _ = forTab {
+        if currentViewController != self {
             self.navigationController?.popViewControllerAnimated(true)
```

Comment:
In the case of this, yes. There are 3 states.

1. The BVC is the top most view controller, in which case we dismiss nothing
2. The Tabs tray is open, in which case that is the top most view controller and we must dismiss it to get to BVC
3. The settings panel is open, in which case that is the presentedViewController, which we dismiss before checking whether or not the topmost view controller is BVC. If not, we are in situation 2.

As yet, there is not another situation where the topmost view controller is more than 1 step away from BVC, or where there are more than 1 presentedViewControllers at a time.